File: 2021--蜜獾算法（Honey Badger Algorithm，HBA）/PYTHON/main.py
from matplotlib import pyplot as plt
from utils import HBA

dim = 30
T = 15
Lb = -10
Ub = 10
N = 30
[xmin, fmin, CNVG] = HBA(dim, Lb, Ub, T, N)

plt.plot(range(len(CNVG)),CNVG,label = 'HBA')
plt.xlim([0,T])
plt.title('Convergence curve')
plt.xlabel('Iteration')
plt.ylabel('Best fitness obtained so far')
plt.legend()
plt.show()

print('The best location= ', xmin)
print('The best fitness score = ', fmin)

File: 2021--蜜獾算法（Honey Badger Algorithm，HBA）/MATLAB/main.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%  Honey Badger Algorithm source code 
%  paper:
%     Hashim, Fatma A., Essam H. Houssein, Kashif Hussain, Mai S. %     Mabrouk, Walid Al-Atabany. 
%     "Honey Badger Algorithm: New Metaheuristic Algorithm for %  %     Solving Optimization Problems." 
%     Mathematics and Computers in Simulation, 2021.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%

clc;clear all;
close all;
fitfun = @sumsqu;
dim=30;
T=1000;
Lb=-10;
Ub=10;
N=30;
[xmin,fmin,CNVG]=HBA(fitfun,dim,Lb,Ub,T,N);
figure,
semilogy(CNVG,'r')
xlim([0 T]);
title('Convergence curve')
xlabel('Iteration');
ylabel('Best fitness obtained so far');
legend('HBA')

display(['The best location= ', num2str(xmin)]);
display(['The best fitness score = ', num2str(fmin)]);

        

